### ISSUE TITLE
Error: (commonjs plugin) Parenthesized pattern

### ISSUE THREAD

**Issue by DanielJoyce** [rollup/rollup-plugin-commonjs#165]
Another weird one that just cropped up and broke the build today. Man is npm a mess, I can only assume something changed somewhere, maybe in a library the commonjs plugin uses and somehow that got pulled in even though common js version is nailed down.

Can't wait till yarn git support matures.

Anyways, this is the error

Directory names changed to protect the innocent...

`Error: (commonjs plugin) Parenthesized pattern (57:2) in $HOME/git/PROJECT_DIR/node_modules/faker/vendor/password-generator.js`

This is the file 

``` javascript

/*
 * password-generator
 * Copyright(c) 2011-2013 Bermi Ferrer <bermi@bermilabs.com>
 * MIT Licensed
 */
(function (root) {

  var localName, consonant, letter, password, vowel;
  letter = /[a-zA-Z]$/;
  vowel = /[aeiouAEIOU]$/;
  consonant = /[bcdfghjklmnpqrstvwxyzBCDFGHJKLMNPQRSTVWXYZ]$/;


  // Defines the name of the local variable the passwordGenerator library will use
  // this is specially useful if window.passwordGenerator is already being used
  // by your application and you want a different name. For example:
  //    // Declare before including the passwordGenerator library
  //    var localPasswordGeneratorLibraryName = 'pass';
  localName = root.localPasswordGeneratorLibraryName || "generatePassword",

  password = function (length, memorable, pattern, prefix) {
    var char, n;
    if (length == null) {
      length = 10;
    }
    if (memorable == null) {
      memorable = true;
    }
    if (pattern == null) {
      pattern = /\w/;
    }
    if (prefix == null) {
      prefix = '';
    }
    if (prefix.length >= length) {
      return prefix;
    }
    if (memorable) {
      if (prefix.match(consonant)) {
        pattern = vowel;
      } else {
        pattern = consonant;
      }
    }
    n = Math.floor(Math.random() * 94) + 33;
    char = String.fromCharCode(n);
    if (memorable) {
      char = char.toLowerCase();
    }
    if (!char.match(pattern)) {
      return password(length, memorable, pattern, prefix);
    }
    return password(length, memorable, pattern, "" + prefix + char);
  };


  ((typeof exports !== 'undefined') ? exports : root)[localName] = password;
  if (typeof exports !== 'undefined') {
    if (typeof module !== 'undefined' && module.exports) {
      module.exports = password;
    }
  }

  // Establish the root object, `window` in the browser, or `global` on the server.
}(this));

```

This is the culprit line:

``` javascript

  ((typeof exports !== 'undefined') ? exports : root)[localName] = password;


```

Apparently whatever tool commonjs rollup is using, a new version got pushed out, and it thinks this is a destructuring assignment, which its not. 

UGH

**Comment by DanielJoyce** [comment]
I suspect problem is a package dependency pushed a new rev 

commonjs plugin deps:

```
  "dependencies": {
    "acorn": "^4.0.1",
    "estree-walker": "^0.3.0",
    "magic-string": "^0.19.0",
    "resolve": "^1.1.7",
    "rollup-pluginutils": "^1.5.1"
  },
```

**Comment by DanielJoyce** [comment]
Broke version deps:

`
├─┬ rollup-plugin-commonjs@7.0.0
│ ├── acorn@4.0.5
│ ├── estree-walker@0.3.0
│ ├─┬ magic-string@0.19.0
│ │ └── vlq@0.2.1
│ └─┬ rollup-pluginutils@1.5.2
│   └── estree-walker@0.2.1

`

working version :

`
├─┬ rollup-plugin-commonjs@7.0.0
│ ├── acorn@4.0.4
│ ├── estree-walker@0.3.0
│ ├─┬ magic-string@0.19.0
│ │ └── vlq@0.2.1
│ └─┬ rollup-pluginutils@1.5.2
│   └── estree-walker@0.2.1

`

**Comment by DanielJoyce** [comment]
npm shrinkwrapped and forced acorn 4.0.4 in the shrinkwrap fixes the issue.

**Comment by DanielJoyce** [comment]
4.0.5 has a bug, so I think commonjs package.json should specify a hard dependency on 4.0.4 for now

**Comment by Rich-Harris** [comment]
Apologies for the delay in responding — can this be closed now that Acorn 4.0.8 is out with a fix?
